### ISSUE TITLE
Code checking incorrectly triggered for particular nimbleFunctionList usage

### ISSUE THREAD

**Issue by perrydv** [nimble-dev/nimble#1084]
The following incorrectly generates a warning that `foo` needs to be defined:

`myNimbleFunctionList[[1]]$foo(x)`

where `foo` is a method in the virtual nimbleFunction for `myNimbleFunctionList`.  The following does not generate the warning:

`i <- 1`. `myNimbleFunctionList[[i]]$foo(x)`, nor does

`myNimbleFunctionList[[1]]$run(x)`.

**Comment by paciorek** [comment]
Here is a reprex, which throws a warning about `foo` and `foo2` but not `foo3`. I will look into this.

```
nfbase= nimbleFunctionVirtual(
    name = 'nfbase',
    run = function() {returnType(double(0))},
    methods = list(
        foo = function() { }
    )
)

nfextend = nimbleFunction(
    setup = function() {},
    run = function() {
        returnType(double(0))
        return(3)
    },
    methods = list(
        foo2 = function() {},
         foo3 = function() {}
    )
)

nfmore = nimbleFunction(
    setup = function() {
        fxns = nimbleFunctionList(nfbase)
        fxns[[1]] <- nfextend()
    },
    run = function() {
        returnType(double(0))
        fxns[[1]]$foo()
        fxns[[1]]$foo2()
        i <- 1
        fxns[[i]]$foo3()
        out = fxns[[1]]$run()
        return(out)
    })
```

**Comment by paciorek** [comment]
Notes on walking the parse tree:

```
out1 = nimble:::RparseTree2ExprClasses(quote(foo$cc(7)))
out2 = nimble:::RparseTree2ExprClasses(quote(foo[[2]]$cc(7)))
out3 = nimble:::RparseTree2ExprClasses(quote(foo[[i]]$cc(7)))
out4 = nimble:::RparseTree2ExprClasses(quote(foo[[x+3]]$cc(7)))
out5 = nimble:::RparseTree2ExprClasses(quote(foo[[myfun[[2]]$innercc()]]$cc(7)))
```

When detect `$`, the 2nd arg in the exprClass is the method name.

**Comment by paciorek** [comment]
It looks like this was addressed, probably in PR #1035 but this was never closed.
